tools/benchmark/api-report/benchmark.api.md

```diff
@@ -109,6 +112,16 @@ export interface CustomBenchmark extends BenchmarkTimingOptions {
 // @public (undocumented)
 export type CustomBenchmarkArguments = MochaExclusiveOptions & CustomBenchmark & BenchmarkDescription;
 
+// @alpha
+export interface CustomBenchmarkOptions {
```

Comment:
To better align with the existing code (to reuse it existing docs and make it easier for third parties to use it with text frameworks other than mocha, I think we should have a non-mocha specific subset of this interface (and use  BenchmarkDescription not just "title" to integrate with the existing tagging logic). Also the runBenchmark (which I think we could rename to just run, since it being a benchmark will always be clear from context) could be made to support synchronous functions (see HookFunction for an example of that) which would be nice for ergonomics since most of our unit tests are synchronous.

Then add mocha specific extension of that adding MochaExclusiveOptions, and put as much as possible of the implementation in the non mocha specific code.

Also this type should probably mark its fields readonly.